cpu/fe310/Kconfig

```
@@ -9,6 +9,7 @@ config CPU_ARCH_RISCV
     bool
     select HAS_ARCH_RISCV
     select HAS_PICOLIBC if '$(RIOT_CI_BUILD)' != '1'
+    select MODULE_MALLOC_THREAD_SAFE
```

As I mentioned here https://github.com/RIOT-OS/RIOT/pull/15606#discussion_r543163346 I think we would need to do it like this, only select it when using Kconfig dependency modelling. Wondering why we did not see an error with #15606.
```suggestion
    select MODULE_MALLOC_THREAD_SAFE if TEST_KCONFIG
```